xarray/core/datatree.py

```diff
@@ -1984,6 +1985,63 @@ def compute(self, **kwargs) -> Self:
         new = self.copy(deep=False)
         return new.load(**kwargs)
 
+    def _persist_inplace(self, **kwargs) -> Self:
+        """Persist all Dask arrays in memory"""
+        # access .data to coerce everything to numpy or dask arrays
+        lazy_data = {
+            path: {
+                k: v._data
+                for k, v in node.variables.items()
+                if is_duck_dask_array(v._data)
+            }
+            for path, node in self.subtree_with_keys
+        }
+        flat_lazy_data = {
+            (path, var_name): array
+            for path, node in lazy_data.items()
+            for var_name, array in node.items()
+        }
+        if flat_lazy_data:
+            import dask
+
+            # evaluate all the dask arrays simultaneously
+            evaluated_data = dask.persist(*flat_lazy_data.values(), **kwargs)
```

Comment:
Sure, I just copied dataset persist, which looks like it hasn't been touched since long before the chunkmanager was introduced.

Should we add persist here as a non-abstract method on the chunkmanager classes? Looks pretty straightforward.

Reply:
Yes that would be great

Reply:
Technically it could be a separate PR (pre-requisite to this one) but it's not super important to split them.